include/nbl/video/surface/ISurfaceVK.h

```diff
@@ -17,13 +17,240 @@ class ISurfaceVK : public ISurface
+            result.eotf = asset::EOTF_IDENTITY;
+        } break;
+
+        case VK_COLOR_SPACE_DISPLAY_P3_LINEAR_EXT:
+        {
+            result.primary = asset::ECP_DISPLAY_P3;
+            result.eotf = asset::EOTF_IDENTITY;
+        } break;
+
+        case VK_COLOR_SPACE_DCI_P3_NONLINEAR_EXT:
+        {
+            result.primary = asset::ECP_DCI_P3;
+            result.eotf = asset::EOTF_DCI_P3_XYZ;
+        } break;
+
+        case VK_COLOR_SPACE_BT709_LINEAR_EXT:
+        {
+            result.primary = asset::ECP_SRGB;
+            result.eotf = asset::EOTF_IDENTITY;
+        } break;
+
+        case VK_COLOR_SPACE_BT709_NONLINEAR_EXT:
+        {
+            result.primary = asset::ECP_SRGB;
+            result.eotf = asset::EOTF_SMPTE_170M;
+        } break;
+
+        case VK_COLOR_SPACE_BT2020_LINEAR_EXT:
+        {
+            result.primary = asset::ECP_BT2020;
+            result.eotf = asset::EOTF_IDENTITY;
+        } break;
+
+        case VK_COLOR_SPACE_HDR10_ST2084_EXT:
+        {
+            result.primary = asset::ECP_BT2020;
+            result.eotf = asset::EOTF_SMPTE_ST2084;
+        } break;
+
+        case VK_COLOR_SPACE_DOLBYVISION_EXT:
+        {
+            result.primary = asset::ECP_BT2020;
+            result.eotf = asset::EOTF_SMPTE_ST2084;
+        } break;
+
+        case VK_COLOR_SPACE_HDR10_HLG_EXT:
+        {
+            result.primary = asset::ECP_BT2020;
+            result.eotf = asset::EOTF_HDR10_HLG;
+        } break;
+
+        case VK_COLOR_SPACE_ADOBERGB_LINEAR_EXT:
+        {
+            result.primary = asset::ECP_ADOBERGB;
+            result.eotf = asset::EOTF_IDENTITY;
+        } break;
+
+        case VK_COLOR_SPACE_ADOBERGB_NONLINEAR_EXT:
+        {
+            result.primary = asset::ECP_ADOBERGB;
+            result.eotf = asset::EOTF_GAMMA_2_2;
+        } break;
+
+        case VK_COLOR_SPACE_PASS_THROUGH_EXT:
+        {
+            result.primary = asset::ECP_PASS_THROUGH;
+            result.eotf = asset::EOTF_IDENTITY;
+        } break;
+
+        case VK_COLOR_SPACE_EXTENDED_SRGB_NONLINEAR_EXT:
+        {
+            result.primary = asset::ECP_SRGB;
+            result.eotf = asset::EOTF_sRGB;
+        } break;
+
+        case VK_COLOR_SPACE_DISPLAY_NATIVE_AMD: // this one is completely bogus, I don't understand it at all
+        {
+            result.primary = asset::ECP_SRGB;
+            result.eotf = asset::EOTF_UNKNOWN;
+        } break;
+
+        default:
+        {
+            // Todo(achal): Log warning unknown color space
+        } break;
+        }
+
+        return result;
+    }
+
+    static inline ISurface::E_PRESENT_MODE getPresentMode(VkPresentModeKHR in)
+    {
+        if (in <= VK_PRESENT_MODE_FIFO_RELAXED_KHR)
+        {
+            return static_cast<ISurface::E_PRESENT_MODE>(in);
+        }
+        else
+        {
+            // Todo(achal): Log warning unknown present modes
+            return static_cast<ISurface::E_PRESENT_MODE>(0);
```

Comment:
return EPM_COUNT, since 0 static casted is an actual valid value

Reply:
no longer the case